modules/doc/content/getting_started/installation/install_libtorch.md

```diff
@@ -0,0 +1,101 @@
+# Libtorch (Pytorch C++ API)
+
+The way one enables libtorch capabilities in MOOSE depends on
+the operating system (Linux or Mac) and if we use HPC or just a local workstation.
+
+!alert! note
+Before we review the main approaches, it is important to emphasize that
+linking MOOSE with libtorch on Linux machines is not supported if the compiler stack has been built
+using a `libc` version below 2.27 (for `libtorch v 1.8+`)
+or 2.23 (for `libtorch v1.4-1.8`). Furthermore, we do not support `libtorch` versions below
+v1.4. To check your currently used libc version use the following command:
+
+```bash
+ldd --version
+```
+
+!alert-end!
+
+## Setting up the environment
+
+For Mac workstations, the user needs to create a conda environment using the
+instructions [here](installation/conda.md). On Linux machines, however,
+we cannot use the conda packages due to the mismatch between `libc` versions.
+For this reason, given that the system `libc` version allows the linking between
+the two libraries, we need to install `PETSc` and `libmesh` manually. For instructions
+on the installation of these, see [installation/hpc_install_moose.md].
+
+In situations when the `libc` version allows the linking but the compiler stack
+has been compiled with an older `libc` version (HPC machines potentially), we need to build the
+compiler from scratch. For instructions in this process, visit [installation/manual_installation_gcc.md]
+
+## Installing Libtorch
+
+The user can choose from two alternatives when it comes to installing `libtorch`:
+
+- +Install using the script provided in MOOSE:+
+
+  For this, navigate to the MOOSE root directory and execute the following script:
+
+  ```bash
+  ./scripts/setup_libtorch.sh
+  ```
```

Comment:
Should we delete ".zip" as well after we already extract contents?

```
(moose) kongf@contrib % ls
README				exodiff				hit				libtorch			mtwist				pugixml
boost				gtest				json				libtorch-macos-1.8.0.zip	pcre				tinydir
(moose) kongf@contrib % 
```

Reply:
I do think we should delete ".zip", otherwise, we will have more and more

```
(moose) kongf@contrib % ls     
README				exodiff				hit				libtorch			libtorch-macos-1.8.0.zip	pcre				tinydir
boost				gtest				json				libtorch-macos-1.10.0.zip	mtwist	
```

Reply:
In order make the workflow  a bit easier in the future, ` ./scripts/setup_libtorch.sh` should "`rm`" all installed torch world and download and install

Reply:
The only reason I left the .zip file is that if the user want to set up the new installation after doing a clobberall or git clean -xf, they won't have to download it again. It is not a huge issue if you are in the office, but can save some data if you are at home.

Reply:
I added a cleanup step at the beginning of the script, will keep the zip files for now.